cmd/serverless/api_key_test.go

```diff
@@ -103,6 +106,23 @@ func TestExtractRegionFromMalformedPrefixSecretsManagerArnPrefix(t *testing.T) {
 	assert.Error(t, err, "could not extract region from arn: aws:secretsmanager:us-west-2:123456789012:secret:DatadogAPIKeySecret. arn: invalid prefix")
 }
 
+func TestSendAPIKeyToShellSuccess(t *testing.T) {
+	port := testutil.FreeTCPPort(t)
+	hostAndPort := fmt.Sprintf("%s%d", "localhost:", port)
+	listen, err := net.Listen("tcp", hostAndPort)
+	t.Setenv("DD_SERVERLESS_SHELL_PORT", fmt.Sprint(port))
```

Comment:
Is there a reason for doing the `Setenv` there and not just before the last line ?
It seems weird that it's between the call to `Listen` and its error handling.